modin/_compat/io/latest/base_io.py

```diff
@@ -0,0 +1,337 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module houses ``LatestBaseIOCompat`` class.
+
+``LatestBaseIOCompat`` is base mixin for IO classes relying on older pandas.
+"""
+from typing import Optional, Any
+import pickle
+
+import pandas
+import pandas._libs.lib as lib
+from pandas.util._decorators import doc
+
+from modin.utils import _inherit_docstrings
+from ..doc_common import (
+    _doc_default_io_method,
+    _doc_returns_qc,
+    _doc_returns_qc_or_parser,
+)
+
+
+class LatestBaseIOCompat:
+    """Class for basic utils and default implementation of IO functions for older pandas."""
+
+    @classmethod
+    @_inherit_docstrings(pandas.read_parquet, apilink="pandas.read_parquet")
+    @doc(
+        _doc_default_io_method,
+        summary="Load a parquet object from the file path, returning a query compiler",
+        returns=_doc_returns_qc,
+    )
+    def read_parquet(
+        cls, path, engine, columns, storage_options, use_nullable_dtypes, **kwargs
+    ):  # noqa: PR01
+        return cls._read_parquet(
+            path=path,
+            engine=engine,
+            columns=columns,
+            storage_options=storage_options,
+            use_nullable_dtypes=use_nullable_dtypes,
+            **kwargs
+        )
+
+    @classmethod
+    @_inherit_docstrings(pandas.read_csv, apilink="pandas.read_csv")
+    @doc(
+        _doc_default_io_method,
+        summary="Read a comma-separated values (CSV) file into query compiler",
+        returns=_doc_returns_qc_or_parser,
+    )
+    def read_csv(
```

Comment:
Why are there default values for this function, but not for `read_parquet` function above?

Reply:
`read_parquet` is not 100%-corresponding here because of the missed default values, is it?

Reply:
@anmyachev Yes, the defaults changed between versions.

Reply:
wait, I was mixing things... this is our implementation (internal), not visible to the user

Reply:
Default values are set in files `modin/_compat/pandas/latest/io.py` and `modin/_compat/pandas/py36/io.py`. Then they are passed to various IO classes via dispatcher (`BaseIO`, `PandasOnRayIO`). Why again specify default values here (in `modin/_compat/io/latest/base_io.py`)?

Reply:
Again, those are different `read_csv`-s 🙃 
Those living in `modin/_compat/pandas_api` (renamed) are eventually exposed to our end user (as `modin.pandas.read_csv` etc.).

And these ones (in `modin/_compat/core` - renamed, too) are used internally, and, specifically, `BaseIO` is the base class for implementing _storage format_ which defaults to pandas for every operation (hence we need two versions of it - as we support two versions of pandas).

That said, I guess we can think of dropping the default values in storage format implementations, but this feels like a separate feature which I would rather _not_ touch in this PR.

Reply:
As far as I remember, we do not use functions from `BaseIO` class as entry points, this is just one of the intermediate call points, and in this case it is not very clear the need to define default values ​​here.

This was before your pull request, it's just that now there are twice as many such things and so I decided to think about the need for this code :)

Reply:
this could be refactored separately, this PR is already too big and took too long 😄

Reply:
agree